packages/site-admin/src/router/components/link/stories/index.stories.tsx

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { __experimentalVStack as VStack } from '@wordpress/components';
+/**
+ * Internal dependencies
+ */
+import { Link } from '../';
+import { browserHistory, RouterProvider, useLocation } from '../../../';
+import type { Meta, StoryObj } from '@storybook/react';
+import './style.stories.scss';
+
+const meta: Meta< typeof Link > = {
+	title: 'Components/Link',
+	component: Link,
+	tags: [ 'autodocs' ],
+	decorators: [
+		function WithRouterProvider( Story ) {
+			return (
+				<RouterProvider routes={ [] } pathArg="p">
+					<Story />
+				</RouterProvider>
+			);
+		},
+	],
+};
+
+export default meta;
+
+type Story = StoryObj< typeof Link >;
+
+/**
+ * The Link component enables navigation between routes in the application,<br />
+ * maintaining state and avoiding full page reloads.
+ */
+export const Default: Story = {
+	args: {
+		to: '/home',
+		children: 'Homepage',
+		className: 'story-link',
+	},
+};
+
+/**
+ * This story shows how to pass custom state to the Link component using the options prop.<br />
+ * Each time the user clicks the link, a new _lucky number_ is generated and passed along.<br />
+ * You can see how the state is preserved when navigating back and forth.
+ */
+export const PassCustomState: Story = {
+	args: {
+		to: '/home',
+		children: 'Request lucky number!',
+	},
+
+	render: function Template( args ) {
+		useLocation();
+
+		// Pick the location state from the browser history state
+		const { init, history } = ( browserHistory.location.state || {
+			luckyNumber: 0,
+			init: true,
+			history: [],
+		} ) as unknown as {
+			luckyNumber: number;
+			init: boolean;
+			history: number[];
+		};
+
+		let newLuckyNumber = 0;
+
+		/*
+		 * Create a new lucky number only when the user clicks on the link.
+		 */
+		if ( ! init && browserHistory.action !== 'POP' ) {
+			newLuckyNumber = ( ( Math.random() * 100 ) | 0 ) + 1;
+		}
```

Comment:
Is it possible that any re-render of this component is going to create a new lucky number, instead of only when the link is clicked?